### PR TITLE
[zh-cn] Chapter 8 1/2 readme && assignment

### PR DESCRIPTION
**Pull request by swartz-k**
Add  readme && assignment for 8-Reinforcement

### CODE REVIEW

**Comment by jlooper** [comment]
Thank you! Can you find a reviewer in the Discussion boards and we’ll get this added? 🙏

File: 8-Reinforcement/1-QLearning/translations/README.zh-cn.md [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+在上面的算法中，我们没有指定在步骤 2.1 中我们应该如何选择一个动作。如果我们随机选择动作，我们会随机**探索**环境，我们很可能会经常死亡以及探索我们通常不会去的区域。另一种方法是**利用**我们已经知道的 Q-Table 值，从而在状态 *s* 选择最佳动作（具有更高的 Q-Table 值）。然而，这将阻止我们探索其他状态，而且我们可能找不到最佳解决方案。
+
+因此，最好的方法是在探索和开发之间取得平衡。这可以通过选择状态 *s* 的动作来完成，概率与 Q 表中的值成正比。一开始，当 Q-Table 值都相同时，它将对应于随机选择，但是随着我们对环境的了解越来越多，我们将更有可能遵循最佳路线，同时允许智能体选择未探索的路径偶尔。
+
+## Python 实现
+
+我们现在准备实现学习算法。在我们这样做之前，我们还需要一些函数来将 Q-Table 中的任意数字转换为相应动作的概率向量。
+
+1. 创建一个函数`probs()`：
```
Comment (ChungZH) [inline 8-Reinforcement/1-QLearning/translations/README.zh-cn.md:220]: spaces

File: 8-Reinforcement/1-QLearning/translations/README.zh-cn.md [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+            n+=1
+```
+
+执行此算法后，应使用定义每个步骤不同动作的吸引力的值更新 Q 表。我们可以尝试通过在每个单元格上绘制一个向量来可视化 Q-Table，该向量将指向所需的移动方向。为简单起见，我们画一个小圆圈而不是箭头。
+
+<img src="../images/learned.png"/>
+
+## 检查策略
+
+由于 Q-Table 列出了每个状态下每个动作的"吸引力"，因此很容易使用它来定义我们世界中的高效导航。在最简单的情况下，我们可以选择最高Q-Table值对应的action：（代码块9）
```
Comment (ChungZH) [inline 8-Reinforcement/1-QLearning/translations/README.zh-cn.md:266]: spaces

**Comment by jlooper** [comment]
thank you for the review, @ChungZH ! @swartz-k if you can approve those changes I'll approve :)

File: 8-Reinforcement/1-QLearning/translations/README.zh-cn.md [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+    if x==Board.Cell.apple：
+        return goal_reward
+    return move_reward
+```
+
+关于奖励函数的一个有趣的事情是，在大多数情况下，*我们只在游戏结束时才得到实质性的奖励*。这意味着我们的算法应该以某种方式记住最终导致积极奖励的"好"步骤，并增加它们的重要性。同样，所有导致不良结果的举动都应该被劝阻。
+
+## Q-Learning
+
+我们将在这里讨论的一种算法称为**Q-Learning**。在该算法中，策略由称为**Q-Table** 的函数（或数据结构）定义。它记录了给定状态下每个动作的"优点"。
```
Comment (ChungZH) [inline 8-Reinforcement/1-QLearning/translations/README.zh-cn.md:161]: spaces

File: 8-Reinforcement/1-QLearning/translations/README.zh-cn.md [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+现在我们需要运行，探索我们的环境，并学习更好的 Q-Table 值分布，这将使我们能够更快地找到通往苹果的路径。
+
+## Q-Learning 的本质：贝尔曼方程
+
+一旦我们开始移动，每个动作都会有相应的奖励，即理论上我们可以根据最高的即时奖励来选择下一个动作。但是，在大多数情况，此举不会实现我们到达苹果的目标，因此我们无法立即决定哪个方向更好。
+
+> 请记住，重要的不是直接结果，而是我们将在模拟结束时获得的最终结果。
+
+为了解释这种延迟奖励，我们需要使用**[动态编程](https://en.wikipedia.org/wiki/Dynamic_programming)** 的原则，它允许我们递归地思考问题。
```
Comment (ChungZH) [inline 8-Reinforcement/1-QLearning/translations/README.zh-cn.md:183]: 动态规划 而不是 动态编程

File: 8-Reinforcement/1-QLearning/translations/README.zh-cn.md [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+   1. 重复
+        1. 在状态 *s* 选择一个动作 *a*
+        2.通过移动到新状态*s'*来执行动作
+        3.如果我们遇到游戏结束的情况，或者总奖励太少——退出模拟  
+        4. 计算新状态下的奖励*r*
+        5. 根据 Bellman 方程更新 Q-Function： *Q(s,a)* ← *(1-α)Q(s,a)+α(r+γ max<sub>a'</sub>Q( s',a'))*
+        6. *s* ← *s'*
+        7. 更新总奖励并减少α。
+
+##利用与探索
```
Comment (ChungZH) [inline 8-Reinforcement/1-QLearning/translations/README.zh-cn.md:210]: space

File: 8-Reinforcement/1-QLearning/translations/README.zh-cn.md [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+            n+=1
+```
+
+执行此算法后，应使用定义每个步骤不同动作的吸引力的值更新 Q 表。我们可以尝试通过在每个单元格上绘制一个向量来可视化 Q-Table，该向量将指向所需的移动方向。为简单起见，我们画一个小圆圈而不是箭头。
+
+<img src="../images/learned.png"/>
+
+## 检查策略
+
+由于 Q-Table 列出了每个状态下每个动作的"吸引力"，因此很容易使用它来定义我们世界中的高效导航。在最简单的情况下，我们可以选择最高Q-Table值对应的 action：（代码块9）
```
Comment (ChungZH) [inline 8-Reinforcement/1-QLearning/translations/README.zh-cn.md:266]: space

File: 8-Reinforcement/1-QLearning/translations/README.zh-cn.md [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+
+## 🚀挑战
+
+> **任务 1：** 修改 `walk` 函数，将路径的最大长度限制为一定的步数（比如 100），并观察上面的代码时不时返回这个值。
+
+> **任务 2：** 修改 `walk` 函数，使其不会回到之前已经去过的地方。这将防止 `walk` 循环，但是，代理仍然可能最终"被困"在它无法逃脱的位置。
+
+## 导航
+
+更好的导航策略是我们在训练期间使用的，它结合了利用和探索。在这个策略中，我们将以一定的概率选择每个动作，与 Q-Table 中的值成比例。这种策略可能仍会导致代理返回到它已经探索过的位置，但是，正如您从下面的代码中看到的，它会导致到达所需位置的平均路径非常短（请记住，`print_statistics` 运行模拟100次）：（代码块10）
```
Comment (ChungZH) [inline 8-Reinforcement/1-QLearning/translations/README.zh-cn.md:288]: space

File: 8-Reinforcement/2-Gym/translations/README.zh-cn.md [hunk 8]
```diff
@@ -0,0 +1,341 @@
...
+1. 获取这些数字的最小值和最大值：（代码块 5）
+
+    ```python
+    print(env.observation_space.low)
+    print(env.observation_space.high)
+    ```
+
+    您可能还注意到，每个模拟步骤的奖励值始终为 1。这是因为我们的目标是尽可能长时间地生存，即在最长的时间内将杆保持在合理的垂直位置。
+
+    ✅ 事实上，如果我们设法在 100 次连续试验中获得195的平均奖励，则认为 CartPole 问题已解决。
```
Comment (ChungZH) [inline 8-Reinforcement/2-Gym/translations/README.zh-cn.md:121]: space

File: 8-Reinforcement/2-Gym/translations/README.zh-cn.md [hunk 9]
```diff
@@ -0,0 +1,341 @@
...
+    Q = {}
+    actions = (0,1)
+    
+    def qvalues(state):
+        return [Q.get((state,a),0) for a in actions]
+    ```
+
+    这里我们还定义了一个函数 `qvalues()`，它返回对应于所有可能操作的给定状态的 Q-Table 值列表。如果该条目不存在于 Q-Table 中，我们将返回 0 作为默认值。
+
+## 让我们开始Q-Learning
```
Comment (ChungZH) [inline 8-Reinforcement/2-Gym/translations/README.zh-cn.md:194]: space

File: 8-Reinforcement/2-Gym/translations/README.zh-cn.md [hunk 10]
```diff
@@ -0,0 +1,341 @@
...
+
+    这里，`alpha` 是**学习率**，它定义了我们应该在每一步调整 Q-Table 的当前值的程度。在上一课中，我们从 1 开始，然后在训练期间将 `alpha` 减小到较低的值。在这个例子中，为了简单起见，我们将保持它不变，您可以稍后尝试调整 `alpha` 值。
+
+    `gamma` 是**折扣因素**，它表明我们应该在多大程度上优先考虑未来的奖励而不是当前的奖励。
+
+    `epsilon` 是**探索/开发因素**，它决定了我们是否应该更喜欢探索而不是开发，反之亦然。在我们的算法中，我们将在 `epsilon` 百分比的情况下根据 Q-Table 值选择下一个动作，在剩余的情况下，我们将执行随机动作。这将使我们能够探索我们以前从未见过的搜索空间区域。
+
+    ✅ 在平衡方面 - 选择随机动作（探索）将作为朝着错误方向的随机一拳，杆子必须学习如何从这些"错误"中恢复平衡
+
+###改进算法
```
Comment (ChungZH) [inline 8-Reinforcement/2-Gym/translations/README.zh-cn.md:215]: space

File: 8-Reinforcement/2-Gym/translations/README.zh-cn.md [hunk 11]
```diff
@@ -0,0 +1,341 @@
...
+
+- **接近我们的目标**。我们非常接近实现在连续 100 多次模拟运行中获得 195 个累积奖励的目标，或者我们可能真的实现了！即使我们得到更小的数字，我们仍然不知道，因为我们平均超过 5000 次运行，而在正式标准中只需要 100 次运行。
+  
+- **奖励开始下降**。有时奖励开始下降，这意味着我们可以"破坏"Q-Table 中已经学习到的值，这些值会使情况变得更糟。
+
+如果我们绘制训练进度图，则这种观察会更加清晰可见。
+
+## 绘制训练进度
+
+在训练期间，我们将每次迭代的累积奖励值收集到`奖励`向量中。以下是我们根据迭代次数绘制它时的样子：
```
Comment (ChungZH) [inline 8-Reinforcement/2-Gym/translations/README.zh-cn.md:274]: `奖励` 为何要翻译？

File: 8-Reinforcement/1-QLearning/translations/README.zh-cn.md [hunk 12]
```diff
@@ -0,0 +1,320 @@
...
+    return move_reward
+```
+
+关于奖励函数的一个有趣的事情是，在大多数情况下，*我们只在游戏结束时才得到实质性的奖励*。这意味着我们的算法应该以某种方式记住最终导致积极奖励的"好"步骤，并增加它们的重要性。同样，所有导致不良结果的举动都应该被劝阻。
+
+## Q-Learning
+
+我们将在这里讨论的一种算法称为**Q-Learning**。在该算法中，策略由称为**Q-Table** 的函数（或数据结构）定义。它记录了给定状态下每个动作的"优点"。
+
+之所以称为 Q-Table，是因为将其表示为表格或多维数组通常很方便。由于我们的棋盘尺寸为 "width"x"height"，我们可以使用形状为"宽度"width"x"height"x"len(actions)的numpy数组来表示 Q-Table：（代码块6）
```
Comment (ChungZH) [inline 8-Reinforcement/1-QLearning/translations/README.zh-cn.md:163]: strange

File: 8-Reinforcement/1-QLearning/translations/README.zh-cn.md [hunk 13]
```diff
@@ -0,0 +1,320 @@
...
+
+[![强化学习简介](https://img.youtube.com/vi/lDq_en8RNOo/0.jpg)](https://www.youtube.com/watch?v=lDq_en8RNOo)
+
+> 🎥 点击上图观看 Dmitry 讨论强化学习
+
+## [课前测验](https://white-water-09ec41f0f.azurestaticapps.net/quiz/45/)
+
+## 先决条件和设置
+
+在本课中，我们将用Python代码做一些试验。您应该能够在您的计算机上或云中的某个地方运行本课程中的 JupyterNotebook 代码。
```
Comment (ChungZH) [inline 8-Reinforcement/1-QLearning/translations/README.zh-cn.md:18]: space, `Jupyter Notebook`

File: 8-Reinforcement/1-QLearning/translations/README.zh-cn.md [hunk 14]
```diff
@@ -0,0 +1,320 @@
...
+
+为简单起见，让我们将 Peter 的世界视为一个大小为 `width` x `height` 的方板，如下所示：
+
+![彼得的环境](../images/environment.png)
+
+该板中的每个单元格可以是：
+
+* **地面**，彼得和其他生物可以在上面行走。
+* **水**，你显然不能在上面行走。
+* **树**或**草**，您可以休息的地方。
```
Comment (ChungZH) [inline 8-Reinforcement/1-QLearning/translations/README.zh-cn.md:40]: 你 和 您？

**Comment by ChungZH** [comment]
Too many machine translations are used.

**Pull request closed without merging**